### PR TITLE
POTEL 33 - Record dropped spans when sampling OpenTelemetry spans

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

We now record one dropped span and one dropped transaction, whenever we sample a root transaction in `SentrySampler`.

## :bulb: Motivation and Context

Better client reports

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Review by lbloder — APPROVED** [review]
LGTM 👍

File: sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySampler.java [hunk 1]
```diff
@@ -94,7 +96,18 @@ public SamplingResult shouldSample(
...
+
+    if (!sentryDecision.getSampled()) {
+      scopes
+          .getOptions()
+          .getClientReportRecorder()
+          .recordLostEvent(DiscardReason.SAMPLE_RATE, DataCategory.Transaction);
+      scopes
+          .getOptions()
+          .getClientReportRecorder()
+          .recordLostEvent(DiscardReason.SAMPLE_RATE, DataCategory.Span);
```
Comment (stefanosiano) [inline sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySampler.java:108]: Is there always only one span here, or should we also count the children?

Comment (adinauer) [reply sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySampler.java:108]: sampler is invoked per span, before it's even started, we don't know the number of children it'll eventually have at this point. I can add something to the TODO list to see if we can come up with a number after the fact, but I'm not sure there's a callback that'll allow us to do so.

Comment (adinauer) [reply sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySampler.java:108]: Found a way to record for child spans as well. We now record a lost span when copying a parent sampling decision that is `false`.

**Pull request merged**